From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas R <atoomic@cpan.org>
Date: Mon, 16 Jul 2018 11:26:59 -0600
Subject: [PATCH] skip shadow call when euid > 0 on linux

maybe __linux__ is not restrictive enough
and we could consider using a hint sh file
to enable it only on some specific distro?

Upstream-URL: https://rt.perl.org/Ticket/Display.html?id=133388
---
 SOURCES/perl/pp_sys.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SOURCES/perl/pp_sys.c b/SOURCES/perl/pp_sys.c
index 4cbe323e..5621c65b 100644
--- a/SOURCES/perl/pp_sys.c
+++ b/SOURCES/perl/pp_sys.c
@@ -5549,6 +5549,9 @@ PP(pp_gpwent)
         /* Some AIX setups falsely(?) detect some getspnam(), which
          * has a different API than the Solaris/IRIX one. */
 #   if defined(HAS_GETSPNAM) && !defined(_AIX)
+#      ifdef __linux__
+    if (!PerlProc_geteuid())
+#      endif
         {
             dSAVE_ERRNO;
             const struct spwd * const spwent = getspnam(pwent->pw_name);
